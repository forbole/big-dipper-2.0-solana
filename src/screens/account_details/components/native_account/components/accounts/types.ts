export type AddressesType = string[];

export type AccountsType = {
  stake: AddressesType;
  nonce: AddressesType;
  vote: AddressesType;
  token: AddressesType;
}
