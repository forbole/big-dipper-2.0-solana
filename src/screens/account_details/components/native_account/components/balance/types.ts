export type BalanceType = {
  native: TokenUnit;
  stake: TokenUnit;
  nonce: TokenUnit;
  vote: TokenUnit;
  total: TokenUnit;
}
