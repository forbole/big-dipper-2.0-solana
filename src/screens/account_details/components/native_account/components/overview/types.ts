export type OverviewType = {
  address: string;
}
