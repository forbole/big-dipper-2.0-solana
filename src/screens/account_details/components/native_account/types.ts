export type OverviewType = {
  address: string;
}

export type AccountType = {
  address: string;
}

export type AccountsType = {
  data: {
    [value:number]: AccountType[];
  };
  count: number;
  loading: boolean;
}

export type NativeAccountState = {
  loading: boolean;
  exists: boolean;
  overview: OverviewType;
}
