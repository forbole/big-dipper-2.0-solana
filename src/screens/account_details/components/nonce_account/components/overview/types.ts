export type OverviewType = {
  address: string;
  balance: number;
  authority: string;
  blockhash: string;
  fee: number;
}
