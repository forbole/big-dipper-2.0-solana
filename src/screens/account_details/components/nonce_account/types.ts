export type OverviewType = {
  address: string;
  authority: string;
  balance: TokenUnit;
  blockhash: string;
  fee: number;
}

export type NonceAccountState = {
  loading: boolean;
  overview: OverviewType;
}
