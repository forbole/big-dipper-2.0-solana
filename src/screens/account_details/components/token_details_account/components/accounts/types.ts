export type AccountType = string;

export type AccountsType = {
  addresses: AccountType[]
}
