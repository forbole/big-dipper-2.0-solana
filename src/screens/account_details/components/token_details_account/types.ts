export type OverviewType = {
  address: string;
  mint: string;
  balance: TokenUnit;
  owner: string;
}

export type NonceAccountState = {
  loading: boolean;
  overview: OverviewType;
}
