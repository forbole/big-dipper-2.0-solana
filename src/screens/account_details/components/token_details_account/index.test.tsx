/* eslint-disable */
import React from 'react';
import { createMockClient } from 'mock-apollo-client';
import { ApolloProvider } from '@apollo/client';
import renderer from 'react-test-renderer';
import {
  MockTheme, wait,
} from '@tests/utils';
import { TokenDetailsAccountDetailsDocument } from '@graphql/types';
import TokenAccount from '.';

// ==================================
// mocks
// ==================================
jest.mock('next/router', () => ({
  useRouter: () => ({
    query: {
      address: "9XfsbcUJ8qM4esLtMocrMGLqRPkgYNbcmKLebtBzSjjZ",
    },
  }),
}));

jest.mock('./components', () => ({
  Overview: (props) => <div id="Overview" {...props} />,
}));

const mockData = jest.fn().mockResolvedValue({
  "data": {
    "tokenAccount": [
      {
        "address": "V9nuamcxTwr2iVkVQwidTdzJxuRhtVHsbmkvmroHaAf",
        "mint": "EEhosSQvC2yVDRXRGpkonGFF2WNjtUdzb48GV8TSmhfA",
        "owner": "9T2iv4yXzarKyKgVX4ShNx42NjUAUh63xGohR5B5hjsH",
        "tokenInfo": {
          "decimals": 9
        },
        "tokenUnit": {
          "unitName": "Ammo",
          "logoUrl": "https://arweave.net/rjP_BdMqFsXBWoInFYuVNDdqLzW1xo82egb74WRl3Hc"
        }
      }
    ],
    "tokenAccountBalance": []
  }
});

// ==================================
// unit tests
// ==================================
describe('screen: TokenAccount', () => {
  it('matches snapshot', async () => {
    const mockClient = createMockClient();
    mockClient.setRequestHandler(
      TokenDetailsAccountDetailsDocument,
      mockData,
    );

    let component;
    renderer.act(() => {
      component = renderer.create(
        <ApolloProvider client={mockClient}>
          <MockTheme>
            <TokenAccount />
          </MockTheme>
        </ApolloProvider>,
      );
    });
    await wait();

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
