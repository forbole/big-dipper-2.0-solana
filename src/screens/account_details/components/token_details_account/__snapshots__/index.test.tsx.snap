// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: TokenAccount matches snapshot 1`] = `
<div
  className="makeStyles-overview"
  id="Overview"
  overview={
    Object {
      "address": "9XfsbcUJ8qM4esLtMocrMGLqRPkgYNbcmKLebtBzSjjZ",
      "balance": Object {
        "baseDenom": "",
        "displayDenom": "Ammo",
        "exponent": 9,
        "value": "0.000000000",
      },
      "mint": "EEhosSQvC2yVDRXRGpkonGFF2WNjtUdzb48GV8TSmhfA",
      "mintImageUrl": "https://arweave.net/rjP_BdMqFsXBWoInFYuVNDdqLzW1xo82egb74WRl3Hc",
      "owner": "9T2iv4yXzarKyKgVX4ShNx42NjUAUh63xGohR5B5hjsH",
    }
  }
/>
`;
